src/simplewallet/simplewallet.cpp

```diff
@@ -1304,6 +1398,615 @@ bool simple_wallet::export_raw_multisig(const std::vector<std::string> &args)
   return true;
 }
 
+// MMS ---------------------------------------------------------------------------------------------------
+
+// Methods for commands like "export_multisig_info" usually read data from file(s) or write data
+// to files. The MMS calls now those methods as well, to produce data for messages and to process data
+// from messages. As it would be quite inconvenient for the MMS to write data for such methods to files
+// first or get data out of result files after the call, those methods detect a call from the MMS and
+// expect data as arguments instead of files and give back data by calling 'process_wallet_created_data'.
+bool simple_wallet::called_by_mms()
+{
+  bool by_mms = m_called_by_mms;
+  m_called_by_mms = false;
+  return by_mms;
+}
+
+bool simple_wallet::user_confirms(const std::string &question)
+{
+   std::string answer = input_line(question + tr(" (Y/Yes/N/No): "));
+   return !std::cin.eof() && command_line::is_yes(answer);
+}
+
+bool simple_wallet::choose_mms_processing(const std::vector<tools::processing_data> &data_list, uint32_t &choice)
+{
+  uint32_t choices = data_list.size();
+  if (choices == 1)
+  {
+    choice = 0;
+    return true;
+  }
+  tools::message_store& ms = m_wallet->get_message_store();
+  success_msg_writer() << tr("Choose processing:");
+  std::string text;
+  for (size_t i = 0; i < choices; ++i)
+  {
+    const tools::processing_data &data = data_list[i];
+    text = std::to_string(i+1) + ": ";
+    switch (data.processing)
+    {
+    case tools::message_processing_sign_tx:
+      text += tr("Sign tx");
+      break;
+    case tools::message_processing_send_tx:
+    {
+      tools::message m = ms.get_message_by_id(data.message_ids[0]);
+      if (m.type == tools::message_type_fully_signed_tx)
+      {
+        text += tr("Send the tx for submission to ");
+      }
+      else
+      {
+        text += tr("Send the tx for signing to ");
+      }
+      tools::coalition_member member = ms.get_member(data.receiving_member_index);
+      text += ms.member_to_string(member, 50);
+      break;
+    }
+    case tools::message_processing_submit_tx:
+      text += tr("Submit tx");
+      break;
+    default:
+      text += tr("unknown");
+      break;
+    }
+    success_msg_writer() << text;
+  }
+  
+  std::string line = input_line(tr("Choice: "));
+  if (std::cin.eof() || line.empty())
+  {
+    return false;
+  }
+  bool choice_ok = false;
+  if (line.length() == 1)
+  {
+    char c = line[0];
+    if (isdigit(c) && (c != '0'))
```

Comment:
Just use `atoi`. The string is exactly one byte, so it cannot overflow an `int` on any platform. Also line 1477 will not work with some locales/character encodings.